packages/frontend/src/components/UI/search-domain.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useRef, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { isValidAddress } from '../../lib/utils'
+import { Input } from './input'
+import { ReactComponent as SearchLogo } from '../../assets/icons/search-logo.svg'
+import { Button } from './button'
+
+interface IProps {
+  showBtn: boolean
+  classname: string
+}
+
+function SearchDomain({ showBtn, classname }: IProps) {
+  const navigate = useNavigate()
+  const [searchQuery, setSearchQuery] = useState<string>('')
+  const [inputFocus, setInputFocus] = useState<boolean>(false)
+
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const handleSearch = () => {
```

Comment:
Can implement debounce in search

Reply:
hitting enter or clicking on `Search` will forward to another page, thus the user has to put their whole filecoin domain name and search. Using debouncing doesn't seem necessary here

Reply:
Okay, got it. Not required in that case.